src/services/validationService.ts

```
@@ -0,0 +1,347 @@
+});
+
+/**
+ * Helper function to format validation errors for display
+ */
+export const formatValidationErrors = (validation: ValidationResult): string => {
+  if (validation.isValid) return '';
+
+  let message = validation.errors.join('\n');
+
+  if (validation.warnings && validation.warnings.length > 0) {
+    message += '\n\nWarnings:\n' + validation.warnings.join('\n');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)